fe/fe-core/src/main/java/org/apache/doris/resource/workloadgroup/QueueOfferToken.java

```diff
@@ -17,30 +17,101 @@
 
 package org.apache.doris.resource.workloadgroup;
 
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
 // used to mark QueryQueue offer result
 // if offer failed, then need to cancel query
 // and return failed reason to user client
 public class QueueOfferToken {
 
-    private Boolean offerResult;
+    enum TokenState {
+        ENQUEUE_FAILED,
+        ENQUEUE_SUCCESS,
+        READY_TO_RUN
+    }
+
+    static AtomicLong tokenIdGenerator = new AtomicLong(0);
+
+    private long tokenId = 0;
+
+    private TokenState tokenState;
+
+    private long waitTimeout = 0;
+
+    private long enqueueTime = 0;
```

Comment:
unused now, drop it.

Reply:
No, I will use it in the future. I will display enqueue time in wg tvf or show process list.